tests/vector.rs

```diff
@@ -33,6 +33,109 @@ fn test_from_value() {
     assert_eq!(Vector4::from_value(76.5f64), Vector4::new(76.5f64, 76.5f64, 76.5f64, 76.5f64));
 }
 
+macro_rules! test_op_add {
+    ($VectorN:ident { $($field:ident),+ }, $s:expr, $v:expr) => (
+        // vector + vector ops
+        assert_eq!($v + $v, $VectorN::new($($v.$field + $v.$field),+));
```

Comment:
I mean that the same method is being used to both generate the implementation, and to generate the tests. It's good to have them there for redundancy, but it would be nice to figure out some other testing method. It's not going to hold up me merging though.